mlflow/sklearn/utils.py

```diff
@@ -597,6 +600,12 @@ def _create_child_runs_for_parameter_search(cv_estimator, parent_run, child_tags
     """
     import pandas as pd
 
+    def first_custom_rank_column(df):
```

Comment:
What happens if  a custom rank column is not identified? Perhaps we should pick the first `n` runs and warn the user?

Reply:
It should be identified, that is what we expect from [cv_results_](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV). If a rank column is not identified then I think something bigger has gone wrong in the training and this will be gracefully handled because all of [_create_child_runs_for_parameter_search](https://github.com/mlflow/mlflow/blob/master/mlflow/sklearn/__init__.py#L819) is wrapped in a try catch

Reply:
I don't see `rank_test` explicitly defined in the contract of [cv_results_](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV), though it's used in an example.

Regardless, try/catch is fine as long as we have test coverage here. Can we confirm that we test our handling of child run creation failures for hyperparameter search and introduce coverage if we don't?

Reply:
<img width="884" alt="Screen Shot 2021-05-24 at 12 29 37 PM" src="https://user-images.githubusercontent.com/73549313/119398144-fdd4f480-bc8b-11eb-86e8-4e3594e26609.png">

The example is the only location where they provide docs for `cv_results_`

Reply:
Also we already do have test coverage for graceful failing of [_create_child_runs_for_parameter_search()](https://github.com/mlflow/mlflow/blob/master/tests/sklearn/test_sklearn_autolog.py#L848)

Reply:
Not to be pedantic, but the above text is just an example indicating how a Pandas DataFrame with particular columns (e.g. `rank_test_score`) is translated into a `cv_results_` dictionary; it does not explicitly stipulate that `rank_test_*` metrics are always computed, even if that is the case. Regardless, glad we have coverage! Thank you for auditing!

Reply:
No problem, thanks for the diligent review of my PR!

Reply:
That makes sense, they never explicitly listed what are the contents of `cv_results_` technically.